src/util/nextThought.js

```diff
@@ -0,0 +1,240 @@
+import { store } from '../store.js'
+
+// constants
+import {
+  RANKED_ROOT,
+  ROOT_TOKEN,
+} from '../constants.js'
+
+// util
+import {
+  contextOf,
+  getContexts,
+  getContextsSortedAndRanked,
+  getSortPreference,
+  getThought,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  head,
+  isContextViewActive,
+  isFunction,
+  meta,
+  pathToContext,
+  perma,
+  rootedContextOf,
+  splitChain,
+  nextSibling as thoughtNextSibling,
+  unroot,
+} from '../util.js'
+
+/**
+ * Returns true if the context view is activated and has more than one context
+ * @returns {boolean}
+ */
+const isValidContextView = path =>
+  path.length &&
+  isContextViewActive(pathToContext(path)) &&
+  getContexts(head(path).value).length > 1
+
+/**
+ * Adds the rank of the child thought to every thought in a context.
+ * @param {ContextInfo} contextInfo.   A { context, rank } object returned from getContexts.
+ * @returns {array | null} rankedContext
+ */
+const contextWithThoughtRank = contextInfo => {
+  return contextInfo && contextInfo.context.map(value => {
+    const thought = getThought(value)
+    return { ...(thought || { value }), rank: contextInfo.rank }
```

Comment:
`contextInfo.rank` is just the rank of the head thought. Here it is incorrectly applied to every thought in the path that is returned. This will result in an incorrect `thoughtsRanked` passed to `setCursor` in the end.

(Technically I'm not sure if these steps apply to this instance of the problem or the one in `getPathFromContextChain` mentioned below. Either way it demonstrates the issue.)

## Steps to Reproduce

Create the following thoughts from top to bottom:

  - a (rank: 0)
    - b
      - c
        - x
        - y
  - d (rank: 1)
  - e (rank: 2)
    - y

Set the cursor on `a.b.c.y~.c` and hit `cursorDown` to move the cursor to `a.b.c.y~.e`.

## Current Behavior

`nextThought` returns: 

```js
[
  { value: 'e', rank: 1 },
  { value: 'y', rank: 1 }
]
```

## Expected Behavior

```js
[
  { value: 'e', rank: 2 },
  { value: 'y', rank: 1 }
]
```